### PR TITLE
[W-11911033] Change section anchors to copy link button

### PR DESCRIPTION
**Pull request by jdomingu**
Experimented with multiple options including the proposals in the work item. In the end, I matched the Salesforce Docs functionality:
<https://sfdocs-help.herokuapp.com/docs/success/sfdocs/guide/prerequisites.html#git-and-github>

Here's what that looks like:

On hover:
Screenshot 2023-10-20 at 9 46 23 AM

On click:
Screenshot 2023-10-20 at 9 46 29 AM

Testing:

* Confirmed copying to clipboard succeeds.
* Confirmed display of tooltips on mobile, medium, and large screens.
* Confirmed keyboard navigation works as expected. (You can tab to focus and pressing enter copies to clipboard.)

### CODE REVIEW

**Comment by cheungaryk** [comment]
I am going to do a local build, but in the meantime, the branch is out-of-date with the base branch, and there are some linting errors that are failing this build. To see those, you can either look at the Jenkins job or run `npm run bundle` (or `npm run format:bundle`) locally.

File: src/js/23-copy-link-button.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    document.querySelectorAll('.anchor').forEach((anchor) => {
+      const parentHeading = anchor.parentElement
+      const headerText = parentHeading.textContent
+      if (headerText) {
+        const copyLinkBtn = createCopyLinkBtn(anchor.href)
+        parentHeading.replaceChild(copyLinkBtn, anchor);
+      }
+    })
+  }
+  const addCopyLinkTooltips = () => {
```
Comment (cheungaryk) [inline src/js/23-copy-link-button.js:35]: Add a blank line before this one.

**Review by cheungaryk — COMMENTED** [review]
Validated via local build. This removes the original anchor functionality (jump to the heading + append anchor to URL) with replacement. The functionality of the right TOC is kept as-is.

Once the linting errors are fixed, I can approve.

**Comment by jdomingu** [comment]
Awesome. Thanks for the quick turnaround. I just pulled from main and will get on the linting errors soon...

**Comment by jdomingu** [comment]
Thanks again! Ran `npm run format:bundle` and committed.

**Review by cheungaryk — APPROVED** [review]
LGTM

**Pull request merged**